src/main/java/org/apache/commons/math4/util/MathArrays.java

```diff
@@ -73,16 +73,16 @@ private MathArrays() {}
          * @param array Array to operate on.
          * @return the result of the operation.
          */
-        double evaluate(double[] array);
+        double evaluate(double @MinLen(1) [] array);
         /**
          * @param array Array to operate on.
          * @param startIndex Index of the first element to take into account.
          * @param numElements Number of elements to take into account.
          * @return the result of the operation.
          */
-        double evaluate(double[] array,
-                        int startIndex,
-                        int numElements);
+        double evaluate(double @MinLen(1) [] array,
+                        @IndexFor("#1") int startIndex,
```

Comment:
Related to the comment above: Could `startIndex` be `@IndexOrHigh`? That would make `0` a valid value for `startIndex` even if `array` is empty.
In general: for methods which take an array, an index and a length, calling it like `method(array, array.length, 0)` is usually valid, unless the method really requires the range to be non-empty.